src/plugins/chapters/index.js

```diff
@@ -85,23 +85,38 @@ const ChaptersPlugin = (function () {
     }
   };
 
+  ChaptersPlugin.prototype.getAutoChaptersFileUrl = function getAutoChaptersFileUrl() {
+    const cloudName = this.player.cloudinary.cloudinaryConfig().cloud_name;
+    const currentPublicId = this.player.cloudinary.currentPublicId();
+
+    if (!cloudName || !currentPublicId) {
+      return null;
+    }
+
+    return `https://res.cloudinary.com/${cloudName}/video/upload/v1/${currentPublicId}-chapters.vtt`;
+  };
+
   /**
    * Bootstrap the plugin.
    */
   ChaptersPlugin.prototype.initializeChapters = function initializeChapters() {
-    if (this.options.url) {
+    const urlByNameFlagProvided = typeof this.options.byName === 'boolean';
+    const chaptersUrlProvided = !!this.options.url;
+    const chaptersUrl = this.options.byName === true ? this.getAutoChaptersFileUrl() : this.options.url;
```

Comment:
I would give higher priority to url, this is also easier to read.
```suggestion
    const chaptersUrl = this.options.url ?? this.getAutoChaptersFileUrl();
```

Reply:
no longer the case - this.options can be bool